### ISSUE TITLE
Cannot resubmit MFA code 

### ISSUE THREAD

**Issue by rpatel8del** [aws-amplify/amplify-swift#2615]
### Describe the bug

Once `confirmSignIn` is called with the wrong code, the library doesn't allow the user to resubmit with the right code. (Using SMS MFA)

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Call SignIn
2. Call Confirm SignIn with the wrong code
3. Returns `AuthError` - invalid code
4. Call Confirm SignIn with the right code
5. Returns `AuthError` - invalid code
```

### Expected behavior

Users should be allowed to resubmit the right code

### Amplify Framework Version

2.2.0

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

5.7

### CLI version

2.6.0

### Xcode version

14

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>
{
    "AuthState.configured" =     {
        "AuthenticationState.signingIn" =         {
            "SignInState.resolvingChallenge" =             {
                "SignInChallengeState.error" =                 {
                    challenge = "AWSCognitoIdentityProvider.CognitoIdentityProviderClientTypes.ChallengeNameType.smsMfa";
                    error = "AWSCognitoAuthPlugin.SignInError.service(error: ClientRuntime.SdkError<AWSCognitoIdentityProvider.RespondToAuthChallengeOutputError>.client(ClientRuntime.ClientError.retryError(ClientRuntime.SdkError<AWSCognitoIdentityProvider.RespondToAuthChallengeOutputError>.service(AWSCognitoIdentityProvider.RespondToAuthChallengeOutputError.codeMismatchException(AWSCognitoIdentityProvider.CodeMismatchException(_headers: Optional(Date: Fri, 16 Dec 2022 02:30:55 GMT, \nContent-Type: application/x-amz-json-1.1, \nConnection: keep-alive, \nx-amzn-ErrorType: CodeMismatchException:, \nContent-Length: 87, \nx-amzn-RequestId: 93b968af-6a05-4d44-942b-97f04593603f, \nx-amzn-ErrorMessage: Invalid code or auth state for the user.), _statusCode: Optional(http_status_400), _message: Optional(\"Invalid code or auth state for the user.\"), _requestID: Optional(\"93b968af-6a05-4d44-942b-97f04593603f\"), _retryable: false, _isThrottling: false, _type: ClientRuntime.ErrorType.client, message: Optional(\"Invalid code or auth state for the user.\"))), \nStatus Code: http_status_400 \n Content-Length: 87, \nConnection: keep-alive, \nx-amzn-RequestId: 93b968af-6a05-4d44-942b-97f04593603f, \nx-amzn-ErrorMessage: Invalid code or auth state for the user., \nDate: Fri, 16 Dec 2022 02:30:55 GMT, \nx-amzn-ErrorType: CodeMismatchException:, \nContent-Type: application/x-amz-json-1.1)), nil))";
                    username = "3i****v1";
                };
                challengeType = "AWSCognitoAuthPlugin.AuthChallengeType.smsMfa";
                signInMethod = "AWSCognitoAuthPlugin.SignInMethod.apiBased(AWSCognitoAuthPlugin.AuthFlowType.userSRP)";
            };
        };
        "AuthorizationState.signingIn" =         {
        };
    };
}
AuthError: Invalid code or auth state for the user.
Recovery suggestion: Retry with a valid code
Caused by:
codeMismatch
</details>
```

### Is this a regression?

Yes

### Device

iPhone 14 pro

### iOS Version

iOS 16

### Additional context

![MicrosoftTeams-image (8)](https://user-images.githubusercontent.com/102267573/208016104-8d852eae-e02b-43f1-8450-ef7f3cebb6db.png)

**Comment by royjit** [comment]
@rpatel8del Thank you reporting this issue, I have created a PR that would fix this - https://github.com/aws-amplify/amplify-swift/pull/2617/. Please take a look and let us know if this solves the issue.

**Comment by royjit** [comment]
This fix has been released in the latest version 2.2.1 https://github.com/aws-amplify/amplify-swift/releases/tag/v2.2.1 Please take a look and let us know if you are still facing it.
